File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/core/command/FuelCapacitySetCmd.kt
package com.syncrhrohealth.bandpass_ble_connection.core.command

import com.syncrhrohealth.bandpass_ble_connection.core.command.builder.CommandBuilder
import com.syncrhrohealth.bandpass_ble_connection.core.enum.CommandCode
import com.syncrhrohealth.bandpass_ble_connection.core.handler.DeviceHandler
import java.nio.ByteBuffer
import java.nio.ByteOrder

/** Data format
 *
 *  float       fuel_capacity
 *
 * */

object FuelCapacitySetCmd {
    fun send(fuelCapacity: Float, handler: DeviceHandler) {
        val buffer = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN)

        buffer.putFloat(fuelCapacity)

        val data = buffer.array()

        val packet = CommandBuilder.build(
            CommandCode.PACKET_CMD_FUEL_CAPACITY_SET,
            data
        )

        handler.write(packet)
    }
}

File: ios/Classes/Core/Command/FuelCapacitySetCmd.swift
//
//  FuelCapacitySetCmd.swift
//  Runner
//
//  Created by MAC on 11/3/25.
//

import Foundation

/// Data format:
/// - `float fuel_capacity` (4 bytes, little-endian)
struct FuelCapacitySetCmd {
    static func send(fuelCapacity: Float, handler: DeviceHandler) {
        var data = Data()
        
        // Convert fuelCapacity (Float) to Data (4 bytes, little-endian)
        let fuelCapacityData = Data.Float32ToData(fuelCapacity, byteOder: .LittleEndian)
        data.append(fuelCapacityData)
        
        // Build the packet with CommandBuilder
        let packet = CommandBuilder.build(commandCode: .packetCmdFuelCapacitySet, data: data)
        
        // Send packet
        handler.write(data: packet)
    }
}
